File: C++/Problems/Problems/problem_2914.cpp
#include "problem_2914.h"

bool problem_2914::test() {
    std::string s = "1001";

    int expected = 2;

    auto result = minChanges(s);

    std::cout << result << std::endl;

    return result == expected;
}

int problem_2914::minChanges(std::string s) {
    int n = s.size();
    int start = 0;
    char curr = s[0];
    int result = 0;
    for (int i = 0; i < n; i++)
    {
        if (s[i] != curr)
        {
            int len = i - start;
            if (len % 2 != 0)
            {
                result++;
                i++;
            }
            start = i;
            if (i < n)
                curr = s[i];
        }
    }

    return result;
}

File: Python/Problems/Problems/problem2914.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem2914(Problem):
    def test(self) -> bool:
        s = "1001"

        expected = 2

        result = self.minChanges(s)

        print(result)

        return result == expected

    def minChanges(self, s: str) -> int:
        n = len(s)
        start = 0
        curr = s[0]
        result = 0
        for i in range(n):
            if s[i] != curr:
                l = i - start;
                if l % 2 != 0:
                    result += 1
                    i += 1
                start = i
                if i < n:
                    curr = s[i]

        return result